nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/server/ApplicationLoaderService.java

```diff
@@ -369,15 +366,15 @@ private void initializeRuntimeDependencies() {
         // lifecycle modules are loaded separately
         if (Boolean.valueOf(app.getDeployProperties().getProperty
             (ServerTags.IS_LIFECYCLE))) {
-            return ImmutableList.of();
+            return Collections.unmodifiableList(Collections.emptyList());
```

Comment:
`Collections.emptyList()` is already not modifiable. Both would throw a `UnsupportedOperationException` on e.g. `add`